test/expect.js

```diff
@@ -1205,6 +1205,27 @@ describe('expect', function () {
       expect(testMap).to.not.have.any.keys([20, 1, {13: 37}]);
       expect(testMap).to.not.have.all.keys([aKey, {'iDoNot': 'exist'}]);
 
+      // Using the same assertions as above but with `.deep` flag instead of using referential equality
```

Comment:
Same thing as mentioned in the assert interface tests about adding a non-deep test to demonstrate strict equality.